doc/reference/multisig.md

```diff
@@ -8,57 +8,230 @@ The multi-signature scripts are written as JSON objects. We have three types of
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "cf223afe150cc8e89f11edaacbbd55b011ba44fbedef66fbd37d8c9d",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "372643e7ef4b41fd2649ada30a89d35cb90b7c14cb5de252e6ce6cb7",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "aa453dc184c5037d60e3fbbadb023f4a41bac112f249b76be9bb37ad",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "6b732c60c267bab894854d6dd57a04a94e603fcc4c36274c9ed75952",
+            "type": "requireSignature"
+        }
+    ],
+    "type": "all"
 }
 ```
 
 
 
 ## any
 
-The "any" key indicates that in order to spend this tx output, we require one corresponding signature from the listed payment key hashes.
+The `"type"` key's value, `"any"`, indicates that in order to spend this tx output, we require one corresponding signature from the listed payment key hashes.
 
 ```json
 {
-    "any": [
-        "d92b712d1882c3b0f75b6f677e0b2cbef4fbc8b8121bb9dde324ff09",
-        "4d780ed1bfc88cbd4da3f48de91fe728c3530d662564bf5a284b5321",
-        "3a94d6d4e786a3f5d439939cafc0536f6abc324fb8404084d6034bf8",
-        "b12e094d1db7c0fba5121f22db193d0060efed8be43654f861bb68ae",
-        "9be49d56442b4b8b16cab4e43e238bbdefc6c803d554c82fcd5facc3",
-        "622be5fab3b5c3f371a50a535e4d3349c942a98cecee93b24e2fd11d"
-    ]
+    "scripts": [
+        {
+            "paymentKeyHash": "d92b712d1882c3b0f75b6f677e0b2cbef4fbc8b8121bb9dde324ff09",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "4d780ed1bfc88cbd4da3f48de91fe728c3530d662564bf5a284b5321",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "3a94d6d4e786a3f5d439939cafc0536f6abc324fb8404084d6034bf8",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "b12e094d1db7c0fba5121f22db193d0060efed8be43654f861bb68ae",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "9be49d56442b4b8b16cab4e43e238bbdefc6c803d554c82fcd5facc3",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "622be5fab3b5c3f371a50a535e4d3349c942a98cecee93b24e2fd11d",
+            "type": "requireSignature"
+        }
+    ],
+    "type": "any"
 }
 ```
 
 ## atLeast
 
-The "atLeast" key indicates that in order to spend this tx output, we require "atLeast" 2 corresponding signatures from the list of payment key hashes. The "required" key indicates the minimum number of signatures we need to spend the tx output.
+The `"type"` key's value, "atLeast", indicates that in order to spend this tx output, we require "atLeast" 2 corresponding signatures from the list of payment key hashes. The "required" key indicates the minimum number of signatures we need to spend the tx output.
```

Comment:
Should back tick all the strings including `"atLeast"` if that's what we're choosing to do.